SingularityUI/app/components/common/modalButtons/RedeployModal.jsx

```diff
@@ -0,0 +1,106 @@
+import React, { Component, PropTypes } from 'react';
+import { Link } from 'react-router';
+import { connect } from 'react-redux';
+import { Alert } from 'react-bootstrap';
+import JSONTree from 'react-json-tree';
+import { JSONTreeTheme } from '../../../thirdPartyConfigurations';
+
+import { SaveDeploy } from '../../../actions/api/deploys';
+
+import FormModal from '../modal/FormModal';
+import SimpleModal from '../modal/SimpleModal';
+import Utils from '../../../utils';
+
+class RedeployModal extends Component {
+  static propTypes = {
+    requestId: PropTypes.string.isRequired,
+    saveDeploy: PropTypes.func,
+    doAfterRedeploy: PropTypes.func,
+    deploy: PropTypes.object
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {};
```

Comment:
It's not a no-op - it sets initial state to be an empty object rather than null, so that a bunch of things in `render()` don't need to check if state exists.